terraform/lambda/dynamodb.tf

```diff
@@ -44,3 +57,31 @@ resource "aws_iam_policy" "access-user-issued-credentials-table" {
     ]
   })
 }
+
+resource "aws_iam_policy" "access-auth-codes-table" {
+  name = "access-auth-codes-table"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Sid = "AccessAuthCodesTable"
+        Action = [
+          "dynamodb:PutItem",
+          "dynamodb:UpdateItem",
+          "dynamodb:BatchWriteItem",
+          "dynamodb:GetItem",
+          "dynamodb:BatchGetItem",
+          "dynamodb:Scan",
+          "dynamodb:Query",
+          "dynamodb:ConditionCheckItem"
```

Comment:
We have the delete method in the new `DataStore` class (which I'm not sure we'll need), but it won't work as we don't have the `dynamodb:DeleteItem` permission in here. I don't think we should add it now, but it's worth noting in case we do need it in future.

Reply:
Ah yeah good catch. Interestingly it did seem to work in localstack however so thats interesting to note that these permissions aren't potentially being used locally? Something to potentially investigate

Reply:
Yeah it'd be interesting to know. perhaps localstacks modeling of AWS perms isn't bob on...